src/cli/serve/serve.js

```diff
@@ -109,6 +110,25 @@ export function applyConfigOverrides(rawConfig, opts, extraCliOptions) {
   if (opts.dev) {
     if (opts.serverless) {
       setServerlessKibanaDevServiceAccountIfPossible(get, set, opts);
+
+      // Load mock identity provider plugin and configure realm if supported (ES only supports SAML when run with SSL)
+      if (opts.ssl) {
+        set('plugins.paths', _.compact([].concat(get('plugins.paths'), MOCK_IDP_PLUGIN_PATH)));
+        set(`xpack.security.authc.providers.saml.${MOCK_IDP_REALM_NAME}`, {
+          order: Number.MAX_SAFE_INTEGER,
+          realm: MOCK_IDP_REALM_NAME,
+          icon: 'user',
+          description: 'Continue as Test User',
+          hint: 'Allows testing serverless user roles',
+        });
+        // Add basic realm since defaults won't be applied when a provider has been configured
+        if (!has('xpack.security.authc.providers.basic')) {
+          set('xpack.security.authc.providers.basic.basic', {
+            order: 0,
+            enabled: true,
+          });
+        }
+      }
```

Comment:
I have no objections to having this code here. 

nit: I just wonder if this should be part of https://github.com/elastic/kibana/blob/main/packages/kbn-cli-dev-mode/src/cli_dev_mode.ts (or any similar dev-only package). To clearly separate it from production code.